src/dftbp/common/schedule.F90

```diff
@@ -177,4 +227,144 @@ subroutine getChunkRanges(groupSize, myRank, globalFirst, globalLast, localFirst
   end subroutine getChunkRanges
 
 
+  !> Calculate the chunk ranges for a given MPI-communicator considerung different workload
+  subroutine getChunkIterWithWorkload(groupSize, myRank, globalFirst, globalLast, workload, chunkIter)
+
+    !> Size of the group over which the chunks should be distributed
+    integer, intent(in) :: groupSize
+
+    !> Rank of the current process
+    integer, intent(in) :: myRank
+
+    !> First element of the range
+    integer, intent(in) :: globalFirst
+
+    !> Last element of the range
+    integer, intent(in) :: globalLast
+
+    !> Workload for each item
+    integer, intent(in) :: workload(:)
+
+    !> The chunk iterator
+    type(TChunkIterator), intent(out) :: chunkIter
+
+    integer :: numIndices, rank, i
+    integer, allocatable :: rankWorkload(:), indices(:)
+
+    allocate(rankWorkload(groupSize))
+    allocate(indices(globalLast - globalFirst + 1))
+
+    rankWorkload(:) = 0
+    indices(:) = 0
+    numIndices = 0
+
+    do i = globalFirst, globalLast
+      rank = minloc(rankWorkload, dim=1)
+      rankWorkload(rank) = rankWorkload(rank) + max(1, workload(i))
```

Comment:
Why `max(1, workload(i))`? Wouldn't that be equivalent to what we have without workload considerations, as `rankWorkload(rank)` will be increased by 1 only?

Reply:
I wanted to make sure that if the workload is zero that the chunked loop is executed anyway, because there might be some code parts not depending on the workload. Think about:
```
do while (chunkIter%hasNextIndex())
  iAtom1 = chunkIter%getNextIndex()
  do iNeigh = 1, nNeighbourSK(iAtom1)
    ! Assume that nNeighbourSK(iAtom1) is zero
  end do
  call someOtherRoutine()
end do
```

Reply:
Sorry, I should not review PRs in the evening. I thought, it is `min()`. It all makes sense of course!